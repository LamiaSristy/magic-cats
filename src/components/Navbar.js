import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../assets/cat.png';
import '../stylesheets/Navbar.scss';

const Navbar = () => (
  <nav className="nav">
    <div className="container">
      <ul className="navItems">
        <li>
          <Link to="/" className="navBrand">
            <img className="navBrandImg" src={logo} alt="Brand logo" width="40" height="45" />
          </Link>
        </li>
        <li className="navLink">
          {' '}
          <Link to="/">Magical Cats</Link>
        </li>
        <li className="navLink">
          {' '}
          <Link to="/AboutUs">AboutUs</Link>
        </li>
      </ul>
    </div>
  </nav>
);

export default Navbar;
