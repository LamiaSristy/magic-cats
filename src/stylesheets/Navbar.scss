@import '~bootstrap/scss/bootstrap.scss';

a {
  color: #ec0b74 !important;
}

.container {
  max-width: 1500px;
  margin: 0 auto;
}

.nav {
  background-color: #212222;
  color: #fff;
  padding: 5px;
  position: relative; 
}

.navItems {
  list-style: none;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  
  font-size: 18px;
}

.navLink {
  margin: 0 10px;  
}

.navBrand {
  display: flex;
  align-items: center;
  font-size: 32px;
  line-height: 0.9;
}

@media (min-width: 568px) {
  .navBrand {
    padding: 0 15px;
  }

  .navLink {
    margin: 0 20px;
  }
}
