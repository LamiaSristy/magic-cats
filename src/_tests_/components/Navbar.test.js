import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../components/Navbar';

const setup = () => shallow(<Navbar />);

describe('Navbar component', () => {
  let component;

  beforeEach(() => {
    component = setup();
  });

  it('Should render', () => {
    const element = component.find('.nav');
    expect(element.length).toEqual(1);
  });

  test('renders the nav in the page', () => {
    const item = component.find('.nav');
    expect(item).toMatchSnapshot();
  });

  test('renders the container in the page', () => {
    const item = component.find('.container');
    expect(item).toMatchSnapshot();
  });

  test('renders the navItems in the page', () => {
    const item = component.find('.navItems');
    expect(item).toMatchSnapshot();
  });

  test('renders the navBrand in the page', () => {
    const item = component.find('.navBrand');
    expect(item).toMatchSnapshot();
  });

  test('renders the navLink in the page', () => {
    const item = component.find('.navLink');
    expect(item).toMatchSnapshot();
  });

  test('renders the navBrandImg in the page', () => {
    const item = component.find('.navBrandImg');
    expect(item).toMatchSnapshot();
  });
});
