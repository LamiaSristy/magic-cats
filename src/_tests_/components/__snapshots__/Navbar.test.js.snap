// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders the container in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 2`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 3`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 4`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 5`] = `ShallowWrapper {}`;

exports[`Navbar component renders the loading in the page 6`] = `ShallowWrapper {}`;

exports[`Navbar component renders the nav in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the navBrand in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the navBrandImg in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the navItems in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the navLink in the page 1`] = `ShallowWrapper {}`;
